Task_1.2/SuffixArray.cs

```diff
@@ -0,0 +1,75 @@
+namespace Task_1._2
+{
+    public class SuffixArray
+    {
+        private string s;
+        private List<int> suffixArray = new();
+
+        public List<int> Array => suffixArray;
+
+        public SuffixArray(string input)
+        {
+            s = input;
+            CreateSuffixArray();
+        }
+
+        private void CreateSuffixArray()
+        {
+            for (var i = 0; i < s.Length; ++i)
+                suffixArray.Add(i);
+            QuickSort(ref suffixArray, 0, suffixArray.Count - 1);
```

Comment:
```suggestion
            suffixArray.Sort(specialIntComparer);
```
Пользуйтесь благами высокоуровневого языка. Вместо рукописной сортировки стоит использовать уже реализованные в библиотеке. Они оптимизированы для внутреннего представления типа данных, с которым они работают. 
(-0.2 балла)

Reply:
```
suffixArray.Sort((i,j) => string.Compare(s[i..] + s[0..i], s[j..] + s[0..j], StringComparison.Ordinal));
```
Примечательно, что сортировка вот с таким компаратором оказывается в десятки раз быстрее в сравнении с сортировкой с компаратором `specialIntComparer`.
Опять же использовать уже существующий компаратор для строк почти всегда эффективнее, чем реализовывать свой.
Проверял на строке длиной 15000 и получил 43 секунды против 1.8 секунды.